### PR TITLE
[17.0][MIG] web_chatter_position: Migration to 17.0

### PR DESCRIPTION
**Pull request by Jp-alitec**
Migrated to V17.

### CODE REVIEW

**Comment by Jp-alitec** [comment]
hey @legalsylvain @chienandalu @CRogos  i have closed the previous PR (#2826) and created this new one following every guideline , can you please review?

**Comment by Rad0van** [comment]
This does not work at all for me. It breaks My Preferences dialog completely and I have a feeling it breaks other dialogs as well. Teste on fresh clean 17.0 It is _compileForm(el, params)_ function breaking it...
image

**Comment by Jp-alitec** [comment]
@Rad0van Oh ! its because of the optional chaining 
(https://github.com/OCA/web/pull/2833/commits/12b405bc65130ff15204bfb73c0f194f05182e27)
if (child.attributes?.name?.value !== "button_box") {
    append(form, this.compileNode(child, params));
}
I have fixed this and code is working as expected but , the optional chaining is not accepted by pre-commit !! as soon as i update this, it fails , what should i do to fix that ?

File: pandoc-3.2-1-amd64.deb
Comment (MihranThalhath) [inline pandoc-3.2-1-amd64.deb:1]: Please remove this

**Review by MihranThalhath — CHANGES_REQUESTED** [review]

File: pandoc-3.2-1-amd64.deb
Comment (Jp-alitec) [reply pandoc-3.2-1-amd64.deb:1]: @MihranThalhath Sorry but what ?

Comment (MihranThalhath) [reply pandoc-3.2-1-amd64.deb:1]: @Jp-alitec pandoc deb file

Comment (Jp-alitec) [reply pandoc-3.2-1-amd64.deb:1]: @MihranThalhath Removed it !

Still
image

Comment (pedrobaeza) [reply pandoc-3.2-1-amd64.deb:1]: And both commits should be squashed together, or there will be 60 MB of diff

**Comment by MihranThalhath** [comment]
image
checked on runboat and this is how it looks on invoice view

File: pandoc-3.2-1-amd64.deb
Comment (Jp-alitec) [reply pandoc-3.2-1-amd64.deb:1]: @pedrobaeza Sqashed / rebase everything according to guidelines

Comment (pedrobaeza) [reply pandoc-3.2-1-amd64.deb:1]: The binary file is still here.

Comment (Jp-alitec) [reply pandoc-3.2-1-amd64.deb:1]: @pedrobaeza Where , sorry its not visible to me !

Comment (pedrobaeza) [reply pandoc-3.2-1-amd64.deb:1]: imagen

Now to fix it you need to undo the removal in this commit, put it on a separate one, and then squash with the previous commit (pre-commit auto-fixes)

**Comment by duyquyen96** [comment]
@Jp-alitec PR https://github.com/OCA/web/pull/2795#issue-2242298187 Does this have any effect on PR?

File: web_chatter_position/static/src/scss/form_controller.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        max-width: none;
+    }
+    .o_form_sheet {
+        max-width: $o-form-view-sheet-max-width;
+        width: 100%;
+        @include media-breakpoint-up(md) {
+            margin: $o-sheet-vpadding * 0.2 auto;
+        }
+    }
+}
```
Comment (ryantran-novobi) [inline web_chatter_position/static/src/scss/form_controller.scss:17]:
```suggestion
.o_form_view {
    .o_form_renderer {
        max-width: none;
    }
    .o_form_sheet_bg {
        max-width: none;

        .o_form_statusbar {
            max-width: $o-form-view-sheet-max-width;
            width: 100%;
            margin: 0px auto;
        }
    }
    .o_form_sheet {
        max-width: $o-form-view-sheet-max-width;
        width: 100%;
        @include media-breakpoint-up(md) {
            margin: $o-sheet-vpadding * 0.2 auto;
        };
        @include media-breakpoint-down(sm) {
            width: unset !important;
        }
    }
}
```

**Review by ryantran-novobi — CHANGES_REQUESTED** [review]
This pull request works great on my end, even with Web Responsive and Odoo Enterprise.

I suggested a few changes just to have the layout better represented on a big monitor.

File: web_chatter_position/static/src/scss/form_controller.scss (hunk 1 above)
Comment (ryantran-novobi) [reply web_chatter_position/static/src/scss/form_controller.scss:17]: ☝️ @Jp-alitec

**Comment by dustinhoanghavi** [comment]
@Jp-alitec In the Invoice view form, I see duplicate chatter, can you fix it? It could be even better. Thank you!
image

**Comment by tva-subteno-it** [comment]
Hi, I tried this module for myself because one my client needs it. It works really well on my end, with no bug. You only have the pre-commit formatting left to do.
What is the stage of progress of this PR?

**Comment by icf20** [comment]
> @Jp-alitec  I see duplicate chatter, can you fix it? 

confirmed

**Comment by trisdoan** [comment]
> > @Jp-alitec  I see duplicate chatter, can you fix it?
> 
> confirmed

Hi @ryantran-novobi, I am working on fixing the duplicate issue and want to include your suggestion. As I tried, I saw no effect on the form view. Could you please elaborate (nicer with images)?

**Comment by trisdoan** [comment]
> > > @Jp-alitec  I see duplicate chatter, can you fix it?
> > 
> > 
> > confirmed
> 
> Hi @ryantran-novobi, I am working on fixing the duplicate issue and want to include your suggestion. As I tried, I saw no effect on the form view. Could you please elaborate (nicer with images)?

Hello everyone, I fixed the issue and prepared this https://github.com/OCA/web/pull/2893

Please help to review

**Comment by BialArno** [comment]
mhm, oficially OCA module hasn't been migrated, but based on OCA module someone has already published their own version, that seems to work in 17.0 (tho not a single word regarding the usage of OCA module, tho code is 1:1 besides minor changes to form compiler).

**Review by CRogos — CHANGES_REQUESTED** [review]
be carefull... the *.deb file is still in the commit history and should be removed from the first commit. Everything in the history will blowup the repository size.

image

image

**Comment by BT-aleonard** [comment]
What's the status of removing the deb from the commit? @Jp-alitec

**Comment by yajo** [comment]
Superseded by https://github.com/OCA/web/pull/2893.

**Pull request closed without merging**